components/omega/src/ocn/Eos.cpp

```diff
@@ -0,0 +1,253 @@
+#include "Eos.h"
+#include "DataTypes.h"
+#include "HorzMesh.h"
+
+namespace OMEGA {
+
+Eos *Eos::DefaultEos = nullptr;
+std::map<std::string, std::unique_ptr<Eos>> Eos::AllEos;
```

Comment:
Are we ever going to have more than one instance? Seems dangerous to do so. I'd prefer just a single instance here. We can always delete/create if needed for unit testing where we want to test both.